singer_sdk/batch.py

```diff
@@ -109,3 +112,110 @@ def get_batches(
                     )
                 file_url = fs.geturl(filename)
             yield [file_url]
+
+
+def _is_pyarrow_available() -> bool:
+    return importlib.util.find_spec("pyarrow") is not None
+
+
+class ParquetBatcher(BaseBatcher):
+    """Parquet Record Batcher."""
+
+    def __init__(self, *args: t.Any, **kwargs: t.Any) -> None:
+        """Creates a ParquetBatcher instance if pyarrow is available.
+
+        Args:
+            args: Class constructor positional arguments.
+            kwargs: Class constructor keyword arguments.
+
+        Raises:
+            ImportError: If pyarrow is missing.
+        """
+        self.is_pyarrow_available = _is_pyarrow_available()
+        if self.is_pyarrow_available:
+            super().__init__(*args, **kwargs)
+        else:
+            err = """The 'pyarrow' package is required to use the 'ParquetBatcher'
+                class. Please install it by running 'poetry install -E pyarrow'."""
+            raise ImportError(
+                err,
+            )
+
+    def get_batches(
+        self,
+        records: t.Iterator[dict],
+    ) -> t.Iterator[list[str]]:
+        """Yield manifest of batches.
+
+        Args:
+            records: The records to batch.
+
+        Yields:
+            A list of file paths (called a manifest).
+        """
+        if not self.is_pyarrow_available:
+            return
+        pa = importlib.import_module("pyarrow")
+        pq = importlib.import_module("pyarrow.parquet")
```

Comment:
Why not import directly, e.g. `import pyarrow as pa`?

Reply:
Good callout, it was a relic from a linting issue I was having. Thank you!